Providers/Scripts/2.6x-2.7x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python
+# ===================================
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See license.txt for license information.
+# ===================================
+
+import socket
+import os
+import sys
+import imp
+import hashlib
+import codecs
+import base64
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+LG = nxDSCLog.DSCLog
+
+global show_mof
+show_mof = False
+CONFIG_PATH = '/etc/opt/microsoft/omsagent/conf/'
+SERVER_ADDRESS = '/var/opt/microsoft/omsagent/run/npmdagent.sock'
+DEST_FILE_NAME = 'npmd_agent_config'
+
+#   [key] string ConfigType;
+#   [write] string ConfigID;
+#   [write] string Contents;
+#   [write,ValueMap{"Present", "Absent"},Values{"Present", "Absent"}] string Ensure;
+#   [write] string ContentChecksum;
+
+def init_vars(ConfigType, ConfigID, Contents, Ensure, ContentChecksum):
+    if ConfigType is not None and ConfigType != '':
+        ConfigType = ConfigType.encode('ascii', 'ignore')
```

Comment:
We support only one ConfigType right now. This field is kept for future use. Which file are you referring to?